building_map_tools/building_map/lift.py

```diff
@@ -10,33 +18,291 @@ def __init__(self, yaml_node, name):
+                                           door_size,
+                                           left_door_pose,
+                                           bitmask='0x02'))
+
+        model_ele.append(generate_joint(f'left_joint',
+                                        'prismatic',
+                                        parent,
+                                        'left_door',
+                                        joint_axis='y',
+                                        lower_limit=-self.width / 2,
+                                        upper_limit=0))
+
+    def generate_door_plugin(self, model_ele, name):
+        plugin_ele = SubElement(model_ele, 'plugin')
+        plugin_ele.set('name', f'{name}_plugin')
+        plugin_ele.set('filename', 'libdoor.so')
+        for param_name, param_value in self.params.items():
+            ele = SubElement(plugin_ele, param_name)
+            ele.text = f'{param_value}'
+        door_ele = SubElement(plugin_ele, 'door')
+        door_ele.set('left_joint_name', 'left_joint')
+        door_ele.set('name', f'{name}')
+        door_ele.set('right_joint_name', 'right_joint')
+        door_ele.set('type', 'DoubleSlidingDoor')
 
 
 class Lift:
-    def __init__(self, yaml_node, name):
+    def __init__(self, yaml_node, name, transform, levels):
         self.name = name
         print(f'parsing lift {name}')
 
         self.reference_floor_name = yaml_node['reference_floor_name']
         self.depth = float(yaml_node['depth'])
         self.width = float(yaml_node['width'])
-        self.x = float(yaml_node['x'])
-        self.y = float(yaml_node['y'])
         self.yaw = float(yaml_node['yaw'])
+        raw_pos = (float(yaml_node['x']), -float(yaml_node['y']))
+        self.x, self.y = transform.transform_point(raw_pos)
+        self.cabin_depth = self.depth - 0.1
+        self.cabin_width = self.width - 0.1
 
-        self.level_names = []
+        # default params
+        self.cabin_mass = 800
+        self.params = {
+            'v_max_cabin': 2.0,
+            'a_max_cabin': 1.2,
+            'a_nom_cabin': 1.0,
+            'dx_min_cabin': 0.001,
+            'f_max_cabin': 25323.0}
+
+        self.level_elevation = {}
+        self.level_doors = {}
         self.door_names = []
+        self.level_names = []
         if 'level_doors' in yaml_node:
-            for level_name, door_name in yaml_node['level_doors'].items():
+            for level_name, door_names in yaml_node['level_doors'].items():
+                self.level_doors[level_name] = door_names
+                self.level_elevation[level_name] = levels[level_name].elevation
+                self.door_names.append(door_names)
                 self.level_names.append(level_name)
-                self.door_names.append(door_name)
 
         self.doors = []
         if 'doors' in yaml_node:
             self.doors = self.parse_lift_doors(yaml_node['doors'])
 
+        # for wall generation
+        self.vertices = \
```

Comment:
The name `vertices` is a bit misleading. For `front` and `back` the array represents the x-coordinate of the corner vertices while for `right` and `left`, it represents the y-coordinate of the vertices. (correct me if I'm wrong). The same conventions is used for `door.side`. Would switching to a list of tuples (x-coord, y-coord) for both places be more intuitive?

Reply:
I intend to make each array represent the 1-dimensional positions of the endpoints on each side. It makes the process of sorting and picking endpoints of walls easier. I think switching to tuples of (x, y) is a bit redundant. Maybe changing the dictionary name to `end_points` will be better?